pkg/controller/storagecluster/reconcile.go

```diff
@@ -958,3 +970,64 @@ fi`
 
 	return job
 }
+
+// inhibitPDBAlert patches the Alertmanager config to inhibit PDB alerts for Rook OSDs
+// Inhibition Rules will be added if there is at least one Storage Cluster and
+// be removed otherwise
+func inhibitPDBAlert(client client.Client, reqLogger logr.Logger) error {
+	reqLogger.Info("Adding inhibition rules for PDB alerts on OSDs")
+	storageClusterList := &ocsv1.StorageClusterList{}
+	err := client.List(context.TODO(), storageClusterList)
+	if err != nil {
+		return fmt.Errorf("couldn't list Storage Clusters. %v", err)
+	}
+	storageClusterLength := len(storageClusterList.Items)
+
+	alertmanagerConfigSecret := &corev1.Secret{}
+	alertmanagerConfigSecretName := "alertmanager-main"
+	alertmanagerConfigSecretNamespace := "openshift-monitoring"
+	err = client.Get(context.TODO(), types.NamespacedName{Name: alertmanagerConfigSecretName, Namespace: alertmanagerConfigSecretNamespace}, alertmanagerConfigSecret)
```

Comment:
@simonpasquier This line returns a "not Found" error, but the same thing works on E2E test (https://github.com/openshift/ocs-operator/pull/633/files#diff-5643ef1ea1266f32e13238304f0cd425R136). Any clue why that might happen?

If it was an RBAC issue, i'd expect it to complain saying I can't "get" secret instead of "not found" while it clearly exists.

Reply:
Hmm, not sure why you'd get a "not found" error...